sw/device/tests/entropy_src_fw_override_test.c

```diff
@@ -0,0 +1,407 @@
+      kDifToggleEnabled));
+
+  // Enable the interrupt handler to provide CSRNG with entropy.
+  irq_external_ctrl(true);
+
+  // Enable CSRNG
+  LOG_INFO("Enabling CSRNG");
+  TRY(dif_csrng_configure(&csrng));
+
+  // Enable EDNs in auto request mode
+  LOG_INFO("Enabling EDNs");
+  TRY(dif_edn_set_auto_mode(&edn0, edn_params0));
+  TRY(dif_edn_set_auto_mode(&edn1, edn_params1));
+
+  LOG_INFO("Entropy complex configured");
+
+  return OK_STATUS();
+}
+
+/**
+ * Configure the entropy source in extract and insert mode and run some entropy
+ * consumers.
+ */
+status_t firmware_override_extract_insert(
+    dif_entropy_src_single_bit_mode_t single_bit_mode,
+    bool bypass_conditioner) {
+  LOG_INFO("==================");
+  LOG_INFO("Configuring single_bit_mode=%u, bypass_conditioner=%d",
+           single_bit_mode, bypass_conditioner);
+
+  entropy_config(single_bit_mode, bypass_conditioner);
+
+  // Generate some random numbers.
+  uint32_t data;
+  for (size_t i = 0; i < 16; i++) {
+    TRY(rv_core_ibex_testutils_get_rnd_data(&rv_core_ibex, kTimeoutUsec,
+                                            &data));
+  }
+
+  // Run an AES encryption and decryption process.
+  dif_aes_transaction_t transaction = {
+      .operation = kDifAesOperationEncrypt,
+      .mode = kDifAesModeEcb,
+      .key_len = kDifAesKey256,
+      .key_provider = kDifAesKeySoftwareProvided,
+      .mask_reseeding = kDifAesReseedPerBlock,
+      .manual_operation = kDifAesManualOperationAuto,
+      .reseed_on_key_change = false,
+      .ctrl_aux_lock = false,
+  };
+
+  TRY(aes_testutils_setup_encryption(transaction, &aes));
+  AES_TESTUTILS_WAIT_FOR_STATUS(&aes, kDifAesStatusOutputValid, true,
+                                kTimeoutUsec);
+  TRY(aes_testutils_decrypt_ciphertext(transaction, &aes));
+
+  // Run KMAC
+  dif_kmac_key_t software_key = {
+      .share0 = {0x43424140, 0x47464544, 0x4B4A4948, 0x4F4E4D4C, 0x53525150,
+                 0x57565554, 0x5B5A5958, 0x5F5E5D5C},
+      .share1 = {0},
+      .length = kDifKmacKeyLen256,
+  };
+  uint32_t output[8];
+
+  TRY(kmac_testutils_config(&kmac, false));
+  TRY(kmac_testutils_kmac(&kmac, kDifKmacModeKmacLen128, &software_key,
+                          /*custom_string=*/NULL, /*custom_string_len=*/0,
+                          /*message=*/"hello", /*message_len=*/6,
+                          ARRAYSIZE(output), output));
+
+  // Run HMAC.
+  dif_hmac_transaction_t hmac_transaction = {
+      .digest_endianness = kDifHmacEndiannessLittle,
+      .message_endianness = kDifHmacEndiannessLittle,
+  };
+
+  TRY(dif_hmac_mode_sha256_start(&hmac, hmac_transaction));
+  TRY(hmac_testutils_push_message(&hmac, (char *)kHmacRefLongKey,
+                                  sizeof(kHmacRefLongKey)));
+  TRY(hmac_testutils_check_message_length(&hmac, sizeof(kHmacRefLongKey) * 8));
+  TRY(dif_hmac_process(&hmac));
+  dif_hmac_digest_t digest;
+  TRY(hmac_testutils_finish_polled(&hmac, &digest));
```

Comment:
Unfortunately HMAC doesn't have an EDN connection. So won't request entropy. Instead, it would e good to run a simple test on OTBN which has 2 EDN interfaces :-)

From the `edn_auto_mode.c` test, I think you should be able to use this function `otbn_randomness_test_prepare(&otbn, kOtbnRandomnessIterations);` to load a program that will use quite some entropy on both interfaces. Then you can run it and wait for completion.